src/gui/utils.cpp

```diff
@@ -159,15 +160,22 @@ void Utils::Gui::openFolderSelect(const Path &path)
     }
 
 #ifdef Q_OS_WIN
-    HRESULT hresult = ::CoInitializeEx(nullptr, COINIT_MULTITHREADED);
-    PIDLIST_ABSOLUTE pidl = ::ILCreateFromPathW(reinterpret_cast<PCTSTR>(path.toString().utf16()));
-    if (pidl)
+    auto thread = std::unique_ptr<QThread>(QThread::create([path]()
     {
-        ::SHOpenFolderAndSelectItems(pidl, 0, nullptr, 0);
-        ::ILFree(pidl);
-    }
-    if ((hresult == S_OK) || (hresult == S_FALSE))
-        ::CoUninitialize();
+        if (SUCCEEDED(::CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE)))
+        {
+            PIDLIST_ABSOLUTE pidl = ::ILCreateFromPathW(reinterpret_cast<const wchar_t *>(path.toString().utf16()));
+            if (pidl)
+            {
+                ::SHOpenFolderAndSelectItems(pidl, 0, nullptr, 0);
+                ::ILFree(pidl);
+            }
+
+            ::CoUninitialize();
+        }
+    }));
+    thread->start();
+    thread->wait();
```

Comment:
I think this might block the main thread badly, maybe use the following instead?
```c++
auto thread = QThread::create([path]() {...});
connect(thread, &QThread::finished, thread, &QObject::deleteLater);
thread->start();
```

Reply:
> I think this might block the main thread badly

It shouldn't block the main thread much more than directly executing that code there. At least in Qt it is done this way. Nevertheless, your suggestion looks reasonable to try it out.

Reply:
Done.